src/gui/addnewtorrentdialog.cpp

```diff
@@ -621,32 +636,63 @@ void AddNewTorrentDialog::displayContentTreeMenu(const QPoint &)
     }
 }
 
+void AddNewTorrentDialog::setAddTorrentParams(const BitTorrent::AddTorrentParams &inParams)
+{
+    torrentParams = inParams;
+    
+    if (!torrentParams.savePath.isEmpty())
+        setSavePath(torrentParams.savePath);
+    
+    if (!torrentParams.category.isEmpty()) {
+        int index = ui->categoryComboBox->findText(torrentParams.category);
+        
+        if (index != -1) {
+            ui->categoryComboBox->setCurrentIndex(index);
+        }
+        else {
+            ui->categoryComboBox->addItem(torrentParams.category);
```

Comment:
I thought the reasonable behavior in this case is for it to create a new category. Note that the user can type in any category they want into `categoryComboBox` and if it doesn't exist, qBittorrent will create it. So the command line argument should behave the same way as providing a value for the combo box. This means it's the responsibility of the user to get the category name right, but that's always been the case.